components/sidebar/index.tsx

```diff
@@ -0,0 +1,59 @@
+import dynamic from 'next/dynamic';
+import styled from 'styled-components';
+
+import Box from '@components/box';
+
+import { device } from '@utils';
+
+const Countdown = dynamic(() => import('@components/countdown'), { ssr: false });
+
+const Wrapper = styled.div`
+	display: grid;
+	gap: 1rem;
+	height: 100%;
+`;
+
+const MobileItems = styled.div`
+	display: grid;
+	grid-template-columns: 2fr 1fr;
+	gap: 1rem;
+
+	@media ${device.tablet} {
+		grid-template-columns: auto;
+	}
+`;
+
+const DesktopItems = styled.div`
+	display: none;
+
+	@media ${device.tablet} {
+		display: grid;
+		gap: 1rem;
+	}
+`;
+
+const Sidebar = ({
+	uploaders,
+	data,
+	files,
+	streakers,
+}: {
+	uploaders?: number;
+	data?: number;
+	files?: number;
+	streakers?: number;
```

Comment:
What is the use case that this component could be rendered without any of these props? Or rather, why are they all optional here?

Reply:
Ouch... I really don't know ahhaha
I started this component thinking on getting all the data transformed before and just passed to it, so I could show and hide boxes for mobile/desktop, but on #2 I changed my mind and started to address all inside the componenet.